bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/Capability.java

```diff
@@ -0,0 +1,176 @@
+ */
+package org.openhab.binding.netatmo.internal.handler.capability;
+
+import static org.openhab.binding.netatmo.internal.NetatmoBindingConstants.VENDOR;
+import static org.openhab.core.thing.Thing.*;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.netatmo.internal.api.data.ModuleType;
+import org.openhab.binding.netatmo.internal.api.dto.Device;
+import org.openhab.binding.netatmo.internal.api.dto.Event;
+import org.openhab.binding.netatmo.internal.api.dto.HomeData;
+import org.openhab.binding.netatmo.internal.api.dto.HomeEvent;
+import org.openhab.binding.netatmo.internal.api.dto.HomeStatusModule;
+import org.openhab.binding.netatmo.internal.api.dto.NAHomeStatus.HomeStatus;
+import org.openhab.binding.netatmo.internal.api.dto.NAMain;
+import org.openhab.binding.netatmo.internal.api.dto.NAObject;
+import org.openhab.binding.netatmo.internal.api.dto.NAThing;
+import org.openhab.binding.netatmo.internal.handler.CommonInterface;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link Capability} is the base class for all inherited capabilities
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class Capability {
+    protected final Thing thing;
+    protected final CommonInterface handler;
+
+    protected boolean firstLaunch;
+    protected Map<String, String> properties = Map.of();
+    protected ThingStatus thingStatus = ThingStatus.UNKNOWN;
+    protected @Nullable String thingStatusReason = "";
+
+    Capability(CommonInterface handler) {
+        this.handler = handler;
+        this.thing = handler.getThing();
+    }
+
+    public final void setNewData(@Nullable NAObject newData) {
+        if (newData != null) {
+            beforeNewData();
+            if (newData instanceof HomeData) {
+                updateHomeData((HomeData) newData);
+            }
+            if (newData instanceof HomeStatus) {
+                updateHomeStatus((HomeStatus) newData);
+            }
+            if (newData instanceof HomeStatusModule) {
+                updateHomeStatusModule((HomeStatusModule) newData);
+            }
+            if (newData instanceof Event) {
+                updateEvent((Event) newData);
+            }
+            if (newData instanceof HomeEvent) {
+                updateHomeEvent((HomeEvent) newData);
+            }
+            if (newData instanceof NAThing) {
+                updateNAThing((NAThing) newData);
+            }
+            if (newData instanceof NAMain) {
+                updateNAMain((NAMain) newData);
+            }
+            if (newData instanceof Device) {
```

Comment:
In relation to https://github.com/openhab/openhab-addons/pull/12357#issuecomment-1114051094, and after having read this code a few times, I got the idea that maybe this method could be called more than once for the same module? `Device` is a `NAThing`, so if called two times - with `NAThing` and `Device`, and only of them would be online and the other one offline, this could cause some trouble as both `updateNAThing()` and `updateNADevice()` would be called. I'll try to add some more debug information and verify this theory.